chapters/06-smart-contracts-in-ralph.md

```diff
@@ -2617,16 +2624,17 @@ subscribeContractCreatedEvent({
 }, 0)
 ```
 
-In the `ContractCreatedEvent` event object, `eventIndex` is set to `-1` (a special reserved value for this system event), `contractAddress` is a unique and deterministic value calculated from the contract group and event index, and `fields` contains both the address of the newly created contract and its parent contract id (when applicable). This information is useful for indexers and other off-chain applications to track new contract deployments and their relationships to existing contracts.
+In the `ContractCreatedEvent` event object, `eventIndex` is set to `-1` (a special reserved value for this system event), `contractAddress` is a unique and deterministic value calculated from the contract group and event index, and `fields` contains both the address of the newly created contract and its parent contract ID (when applicable). This information is useful for indexers and other off-chain applications to track new contract deployments and their relationships to existing contracts.
 
 ###### ContractDestroyedEvent
 
 `ContractDestroyedEvent` is emitted when a contract is destroyed:
 
 ```rust
-Contract Foo() {
+Contract Foo(owner: Address) {
   @using(assetsInContract = true)
   pub fn destroy() -> () {
+    checkCaller!(callerAddress!() == owner, 0)
     destroySelf!(callerAddress!())
   }
 }
```

Comment:
When I tried to compile this I got an error that I need to check the caller or to add the annotation. I think it's better to check the caller instead of adding the annotation here since it's a very dangerous function